src/run.rs

```diff
@@ -273,44 +273,37 @@ pub fn run() {
       );
     }
   } else {
-    let name;
-    'outer: loop {
-      for candidate in &["justfile", "Justfile"] {
-        match fs::metadata(candidate) {
-          Ok(metadata) => {
-            if metadata.is_file() {
-              name = *candidate;
-              break 'outer;
-            }
-          }
-          Err(error) => {
-            if error.kind() != io::ErrorKind::NotFound {
-              die!("Error fetching justfile metadata: {}", error)
-            }
-          }
+    match search(
+      env::current_dir()
+        .as_ref()
+        .expect("Error getting current directory"),
```

Comment:
Just should never panic if there is an error that isn't due to a bug, it should always exit with an error code and a useful error message.

`.expect` panics on `Err`, so it can't be used.

Instead, you can do something like:

```
let current_dir = match env::current_dir() {
  Ok(current_dir) => current_dir,
  Err(io_error) => die!("Error getting current dir: {}", io_error),
};

match search(&current_dir) {
...
}
```